pkg/controller/framework/controller.go

```diff
@@ -105,7 +105,10 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 
 // Returns true once this controller has completed an initial resource listing
 func (c *Controller) HasSynced() bool {
-	return c.config.Queue.HasSynced()
+	if c.config.Queue != nil {
```

Comment:
Checking for nil here is probably racy, do you need a lock? Actually how is this possible at all, there may be a bug elsewhere if it's possible to get a Controller pointer with a nil Queue?